doc/benchmark.md

```diff
@@ -0,0 +1,1328 @@
+  kind: HTTPPipeline
+  name: pipeline-demo
+
+- http3: false
+  https: false
+  keepAlive: true
+  keepAliveTimeout: 60s
+  kind: HTTPServer
+  maxConnections: 10240
+  name: server-demo
+  port: 10080
+  rules:
+  - host: ""
+    hostRegexp: ""
+    paths:
+    - backend: pipeline-demo
+      headers: []
+      pathPrefix: /pipeline
+      rewriteTarget: ""
+```
+
+3. HTTPPipeline for baseline
+* In order to compare with NGINX's index.html API. The Pipeline uses ResponseAdaptor to reply an NGINX' index.html like response.(Yeah, replace `nginx` with `easegress` already)
+
+``` yaml
+name: pipeline-demo
+kind: HTTPPipeline
+flow:
+  - filter: rsp
+filters:
+  - name: rsp
+    kind: ResponseAdaptor
+    header:
+      del:
+      set:
+      add:
+    body: >+
+      <!DOCTYPE html>
+      <html>
+      <head>
+      <title>Welcome to easegress!</title>
+      <style>
+          body {
+            width: 35em;
+            margin: 0 auto;
+            font-family: Tahoma, Verdana, Arial, sans-serif;
+          }
+      </style>
+      </head>
+      <body>
+      <h1>Welcome to easegress!</h1>
+      <p>If you see this page, the easegress web server is successfully installed and working. Further configuration is required.</p>
+
+      <p>For online documentation and support please refer to <a href="http://easegress.org/">easegress.org</a>.<br/> Commercial support is available at <a href="http://easegress.com/">easegress.com</a>.</p>
+      <p><em>Thank you for using easegress.</em></p>
+      </body>
+      </html>
+```
+
+**Traefik**
+1. Running binary directly with command `./traefik -c ./traefik.yml` [1]
+2. Static lonfig: traefik.yaml
+``` yaml 
+log:
+  level: INFO
+
+entryPoints:
+  web:
+    address: ":8081"
+
+providers:
+  file:
+    filename: /${file_path}/dynamic_conf.yml
+
+```
+3. Dynamic config: dynamic_conf.yml
+``` yaml
+http:
+  routers:
+    my-router:
+      rule: "PathPrefix(`/pipeline`)"
+      service: foo
+      entryPoints:
+      - web
+
+  services:
+    foo:
+      loadBalancer:
+        servers:
+        - url: "http://${echo_server_ip}:9095"
+```
+
+**Echo Server**
+1. [Source code](https://github.com/megaease/easegress/tree/main/example/backend-service/mirror)
+2. Its logic is accepting HTTP request and printing to the console, in this testing, we only use `9095` port.
+
+
+## Testing 
+### Baseline test
+1. **Echo-server**
+* Loading echo-server directly from `vm02` to `vm03`
+* **Scenario 1**: 50 concurrency/900 requests/2 miniutes limitation/not QPS limitation
+
+``` bash
+
+./hey -n 900   -c 50  -m GET http://${vm03_ip}:9095/pipeline -z 2m    
+
+```
+
+* **Scenario 2**: 100 concurrency/90000 requests/2 miniutes limitation/not QPS limitation
+``` bash
+
+./hey -n 90000  -c 100  -m GET http://${vm03_ip}:9095/pipeline -z 2m   
+
+```
+
+* **Scenario 3**: 120 concurrency/90000 requests/ 2 miniutes limitation/not QPS limitation
+
+``` bash
+
+./hey -n 90000  -c 120  -m GET http://${vm03_ip}:9095/pipeline -z 2m   
```

Comment:
in `hey` command line, when `-z` is specified, `-n` is ignored.

Reply:
Thanks for reminding, but in our test env

``` bash
./hey -n 90000  -c 120 -m GET http://${ip}:8080/index.html -z 2m    
Summary:
 Total:	5.8363 secs
 Slowest:	0.0913 secs
 Fastest:	0.0001 secs
 Average:	0.0075 secs
 Requests/sec:	15420.7851
Response time histogram:
 0.000 [1]	|
 0.009 [59076]	|■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■
 0.018 [22476]	|■■■■■■■■■■■■■■■
 0.027 [6389]	|■■■■
 0.037 [1391]	|■
 0.046 [483]	|
 0.055 [94]	|
 0.064 [43]	|
 0.073 [34]	|
 0.082 [2]	|
 0.091 [11]	|
Latency distribution:
 10% in 0.0004 secs
 25% in 0.0012 secs
 50% in 0.0053 secs
 75% in 0.0114 secs
 90% in 0.0179 secs
 95% in 0.0222 secs
 99% in 0.0343 secs
Details (average, fastest, slowest):
 DNS+dialup:	0.0000 secs, 0.0000 secs, 0.0402 secs
 DNS-lookup:	0.0000 secs, 0.0000 secs, 0.0000 secs
 req write:	0.0001 secs, 0.0000 secs, 0.0436 secs
 resp wait:	0.0036 secs, 0.0001 secs, 0.0618 secs
 resp read:	0.0037 secs, 0.0000 secs, 0.0602 secs
Status code distribution:
 [200]	90000 responses
```
It won't affect the total request number and the testing result,  if we put `-z ` at the beginning, it will ignore the `-n` setting. 
But it does confuse readers. So let me update these commands.